### PR TITLE
ADR0009 Pull Request template and communication guidelines

### PR DESCRIPTION
**Pull request by timgentry**
## What?

I've added a Pull Request template and command to add it to projects.

## Why?

Having a Pull Request (PR) template offers multiple benefits to a development team. It standardises the process of contributing changes, making it easier for both the author and the reviewers to understand the context and impact of the changes.

## How?

This includes a Thor `init` command, template markdown file and documentation for the GitHub pages site.

## Testing?

This commit includes a Thor test.

## Screenshots (optional)

0

## Anything Else?

The documentation references the GitHub communication guidelines with advice on the tone and approach to writing PRs and responding to them.

### CODE REVIEW

**Comment by timgentry** [comment]
@ollietulloch if you're happy with this PR, I'll change the status to approved. Thanks

File: lib/way_of_working/generators/pr_template/init.rb [hunk 1]
```diff
@@ -0,0 +1,21 @@
+# frozen_string_literal: true
+
+require 'thor'
+require 'way_of_working/paths'
+
+module WayOfWorking
+  module Generators
+    module PrTemplate
+      # This generator add the Pull Request template to a project
```
Comment (ollietulloch) [inline lib/way_of_working/generators/pr_template/init.rb:9]: Should this be `This generator adds`?

File: .github/PULL_REQUEST_TEMPLATE/pull_request_template.md [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+## How?
+
+{While the Pull Request diff illustrates the "how" of your changes, it's crucial to highlight major design decisions like choosing a recursive method over a loop. Explaining your reasoning helps reviewers understand your thought process and enables a more insightful review.
+
+e.g. This includes a migration, model and controller for user authentication. I'm using Devise to do the heavy lifting. I ran Devise migrations and those are included here.}
+
+## Testing?
+
+{Including tests along with code updates is essential as it prevents merging code without tests or failing tests, which could pose risks if overlooked. It's equally important to describe how you've tested harder-to-test codes, like infrastructure code, and to communicate any untested conditions or edge cases and their associated risks to the reviewer.}
+
```
[the remark this answers is not in the document]
Comment (timgentry) [reply .github/PULL_REQUEST_TEMPLATE/pull_request_template.md:23]: simples mistake 😄

[the remark this answers is not in the document]
Comment (timgentry) [reply .github/PULL_REQUEST_TEMPLATE/pull_request_template.md:23]: This has been corrected.

File: lib/way_of_working/templates/.github/PULL_REQUEST_TEMPLATE/pull_request_template.md [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+## How?
+
+{While the Pull Request diff illustrates the "how" of your changes, it's crucial to highlight major design decisions like choosing a recursive method over a loop. Explaining your reasoning helps reviewers understand your thought process and enables a more insightful review.
+
+e.g. This includes a migration, model and controller for user authentication. I'm using Devise to do the heavy lifting. I ran Devise migrations and those are included here.}
+
+## Testing?
+
+{Including tests along with code updates is essential as it prevents merging code without tests or failing tests, which could pose risks if overlooked. It's equally important to describe how you've tested harder-to-test codes, like infrastructure code, and to communicate any untested conditions or edge cases and their associated risks to the reviewer.}
+
```
[the remark this answers is not in the document]
Comment (timgentry) [reply lib/way_of_working/templates/.github/PULL_REQUEST_TEMPLATE/pull_request_template.md:23]: This has been corrected.

**Review by ollietulloch — APPROVED** [review]
